src_kt/javatests/arcs/util/GuardTest.kt

```diff
@@ -0,0 +1,55 @@
+package arcs.util
+
+import arcs.arcs.util.guardWith
+import arcs.testutil.assertThrows
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** Tests for the [guardWith]-generated property delegate: [GuardDelegate]. */
+@ExperimentalCoroutinesApi
+@RunWith(JUnit4::class)
+class GuardTest {
+    class RequiresLocking(initialValue: Int = 0) {
+        val mutex = Mutex()
```

Comment:
Right here you could put @GuardedBy("this.mutex")? Would it work? Shame it would violate DRY.

Reply:
Yeah :/